optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverJob.java

```diff
@@ -38,17 +38,17 @@
     private final DefaultSolverManager<Solution_, ProblemId_> solverManager;
     private final ProblemId_ problemId;
     private final Solver<Solution_> solver;
-    private final Function<ProblemId_, Solution_> problemFinder;
-    private final Consumer<Solution_> finalBestSolutionConsumer;
+    private final Function<ProblemId_, ? extends Solution_> problemFinder;
+    private final Consumer<? super Solution_> finalBestSolutionConsumer;
     private final BiConsumer<ProblemId_, Throwable> exceptionHandler;
 
     private volatile SolverStatus solverStatus;
     private Future<Solution_> future;
 
     public DefaultSolverJob(DefaultSolverManager<Solution_, ProblemId_> solverManager, ProblemId_ problemId,
             Solver<Solution_> solver,
-            Function<ProblemId_, Solution_> problemFinder,
-            Consumer<Solution_> finalBestSolutionConsumer,
+            Function<ProblemId_, ? extends Solution_> problemFinder,
+            Consumer<? super Solution_> finalBestSolutionConsumer,
```

Comment:
The next line can be `BiConsumer<? super ProblemId_, ? super Throwable> exceptionHandler`.